apps/src/lib/kits/maker/WebSerialPortWrapper.js

```diff
@@ -24,67 +24,66 @@ export default class WebSerialPortWrapper extends EventEmitter {
     // TODO - not sure if this is used in Maker Toolkit yet
   }
 
-  /** Opens the serial port from Micro:Bit and starts reading from port.
-   * We are opening the port here and not reading from or writing to the port at this point in
-   * implementation. Later on when the entire Micro:WebSerial pathway is implemented,
-   * this function may be removed logic added for the MB in the open() function below,
-   * if appropriate.
-   */
-  openMBPort() {
+  // Opens the serial port from Micro:Bit and starts reading from port.
+  async openMBPort() {
     if (this.portOpen) {
       throw new Error(`Requested port is already open.`);
     }
     return this.port
       .open({baudRate: SERIAL_BAUD})
-      .then(() => {
-        this.writer = this.port.writable.getWriter();
-        this.reader = this.port.readable.getReader();
-      })
       .then(() => {
         this.portOpen = true;
         this.emit('open');
+        this.writer = this.port.writable.getWriter();
+        this.reader = this.port.readable.getReader();
+
+        this.readLoop();
       })
+      .then(() => this)
```

Comment:
Also just curious, what's the purpose of this call? Is this so the caller of `openMBPort()` has access to the `port` object?

Reply:
Actually - it's not needed! I'll go ahead and remove.